tools/sdmpredictors/sdmpredictors_list_layers.xml

```diff
@@ -0,0 +1,204 @@
+                   </section>
+                </when>
+            </conditional>
+        </repeat>
+    </inputs>
+    <outputs>
+        <collection name="datasets_output" type="list"
+            label="Datasets list">
+        <discover_datasets pattern="__name_and_ext__" />
+        </collection>
+    </outputs>
+    <tests>
+        <test>
+            <repeat name="list_layers">
+                <conditional name="choose_layer">
+                    <param name="input_layer" value="layers_paleo"/>
+                    <section name="environement">
+                        <param name="terrestrial" value="false"/>
+                        <param name="marine" value="true"/>
+                        <param name="freshwater" value="false"/>
+                    </section>
+                    <section name="advanced_options">
+                        <param name="monthly" value="true"/>
+                        <param name="version" value="0"/>
+                        <param name="model_name" value="All"/>
+                        <param name="epoch" value="All"/>
+                        <param name="years_ago" value="0"/>
+                    </section>
+                </conditional>
+            </repeat>
+            <repeat name="list_layers">
+                <conditional name="choose_layer">
+                    <param name="input_layer" value="layers_future"/>
+                    <section name="environement">
+                        <param name="terrestrial" value="true"/>
+                        <param name="marine" value="false"/>
+                        <param name="freshwater" value="false"/>
+                    </section>
+                    <section name="advanced_options">
+                        <param name="monthly" value="true"/>
+                        <param name="version" value="0"/>
+                        <param name="scenario" value="All"/>
+                        <param name="year" value="0"/>
+                    </section>
+                </conditional>
+            </repeat>
+            <repeat name="list_layers">
+                <conditional name="choose_layer">
+                    <param name="input_layer" value="layers_current"/>
+                    <section name="environement">
+                        <param name="terrestrial" value="false"/>
+                        <param name="marine" value="false"/>
+                        <param name="freshwater" value="true"/>
+                    </section>
+                    <section name="advanced_options">
+                        <param name="monthly" value="true"/>
+                        <param name="version" value="0"/>
+                    </section>
+                </conditional>
+            </repeat>
+            <output_collection name="datasets_output" type="list" count="3">
+                <element name="1_data_paleo" file="1_data_paleo.tsv" ftype="tsv"/>
```

Comment:
you can now use something like this here: https://github.com/galaxyproject/tools-iuc/blob/01e7e0814f8bb4253a92ee4541b555e9af74fb0a/tools/checkm/tree.xml#L88

See also here. https://docs.galaxyproject.org/en/latest/dev/schema.html#tool-tests-test-output-assert-contents